stack/eslint-plugin-packlets/src/DependencyAnalyzer.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type * as ts from 'typescript';
+import * as path from 'path';
+
+import { Path } from './Path';
+import { PacketAnalyzer } from './PackletAnalyzer';
+
+enum RefFileKind {
+  Import,
+  ReferenceFile,
+  TypeReferenceDirective
+}
+
+// TypeScript compiler internal:
+// https://github.com/microsoft/TypeScript/blob/5ecdcef4cecfcdc86bd681b377636422447507d7/src/compiler/program.ts#L541
+interface RefFile {
+  referencedFileName: string;
+  kind: RefFileKind;
+  index: number;
+  file: string;
```

Comment:
Is `file` the source file?

Reply:
It's actually the full path of the referencing file. I'll add some docs.